### DISCUSSION TITLE
Convert macOS NSImage to Avalonia.Media.Imaging.Bitmap

### DISCUSSION THREAD

**Discussion by julianpaulozzi** [AvaloniaUI/Avalonia#17233] [Q&A]
I'm creating a desktop only POC project (Windows, macOS and Linux) where I have to access several native resources. 

I need to get the native system icon for the files types on each platform's.

In `net8.0-windows` project this is handled with `Vanara.Windows.Shell` and https://github.com/AvaloniaUI/Avalonia/discussions/5908#discussioncomment-806242 extension method.

```cs
public Bitmap? GetSystemIcon(string fileNameOrExtension)
{
    var systemIcon = ShellImageList.GetSystemIcon(fileNameOrExtension, ShellImageSize.Large);

    return systemIcon?.ToBitmap()?   // System.Drawing.Bitmap
      .ConvertToAvaloniaBitmap();    // Avalonia.Media.Imaging.Bitmap
}
```

In `net8.0-macos` the `NSWorkspace` is accessible but would have to convert `NSImage` to `Avalonia.Media.Imaging.Bitmap`.

```cs
public Bitmap? GetSystemIcon(string fileNameOrExtension)
{
    var fileExtension = Path.GetExtension(fileNameOrExtension);
    
    // get macOS system file icon
    var icon = NSWorkspace.SharedWorkspace.IconForFileType(fileExtension);
    
    // convert macOS NSImage to Avalonia.Media.Imaging.Bitmap?
}
```
Any way to get this?

**Comment by maxkatz6** [comment]
Not the most efficient approach, but you can use [representationUsingType:properties:](https://developer.apple.com/documentation/appkit/nsbitmapimagerep/1395458-representationusingtype) API on NSImage to encode it into `png`, and then decode with `new Bitmap`.
